packages/docusaurus-1.x/lib/core/nav/HeaderNav.js

```diff
@@ -250,11 +251,15 @@ class HeaderNav extends React.Component {
     headerLinks.forEach(link => {
       if (
         link.doc &&
-        !fs.existsSync(`${CWD}/../${readMetadata.getDocsPath()}/`)
+        !fs.existsSync(path.resolve('../', readMetadata.getDocsPath()))
```

Comment:
Hi @endiliey, I've changed the calls to path.resolve() to be more declarative as you suggested. One thing I noticed is that liveReloadServer.js makes a call to readMetadata.getDocsPath:
https://github.com/facebook/Docusaurus/blob/b4bfa084f1eb21134c4c843492ca8440322a5871/packages/docusaurus-1.x/lib/server/liveReloadServer.js#L20-L21
This pattern seems to refer only to directories which are one level up from the current working directory. How would you suggest changing this? I tried
```js
  gaze(
    [`${path.resolve(CWD, '..', readMetadata.getDocsPath())}/**/*`, '**/*', '!node_modules/**/*'],
```
, but this causes gaze() to produce an error: 
`TypeError: Path must be a string. Received undefined`

Reply:
Your error was because `CWD` was undefined. 

Just add CWD on those lines of code that previously had it.
If it previously doesn't have it, don't add it

Reply:
The error occurs even when I define CWD, or if I use path.resolve() without the CWD parameter. I believe that this line should be changed because `../${readMetadata.getDocsPath()}/**/*` won't work if the user uses an absolute customDocsPath.

Reply:
gaze is a file watcher. And the above is actually a glob pattern. It tries to watch files relative to cwd. 

If absolute path doesnt work, try converting the absolute path relative to cwd。

Try reading v2 code here, we replaced gaze with chokidar. But see that we build the glob pattern correctly even if its an absolutepath https://github.com/facebook/Docusaurus/blob/master/packages/docusaurus/lib/commands/start.js

Also note that glob pattern must be POSIX style otherwise this will break on windows

Reply:
I figured out the issue. When the test in `liveReloadServer.test.js` is run, `liveReloadServer.start()`'s call to `readMetadata.getDocsPath()` returns undefined. So if I add a call to `path.resolve()`, with `readMetadata.getDocsPath()` as a parameter, it causes the error to be thrown, which causes a test failure. Is this an issue with the test, or can we actually expect `readMetadata.getDocsPath()` to return undefined during normal operation?

Reply:
I wasnt aware of the error since the commit is not pushed. Either way, by right readMetadata.getDocsPath should not return undefined 


We should default it to ‘docs’. If the test is wrong, it needs to be fixed

Reply:
Hmm, well the glob pattern `../${readMetadata.getDocsPath()}/**/*` which is being passed to gaze evaluates to `../undefined/**/*` when `liveReloadServer.test.js` is being run , so it's not even doing what it supposed to with the existing glob pattern. Therefore, if I use readMetadata.getDocsPath() as a parameter to a function that expects a string, a test failure occurs. Not sure what to do here, but this should probably be resolved in a different PR.